.github/workflows/_build_publish.yaml

```diff
@@ -0,0 +1,251 @@
+        run: |
+          echo "Extracting image meta for ${{ matrix.image }}"
+          registry=$(make docker/info/registry)
+          tag=$(make build/info/version)
+          echo "tag=${tag}" >> $GITHUB_OUTPUT
+          echo "registry=${registry}" >> $GITHUB_OUTPUT
+          echo "image=${registry}/${{ matrix.image }}:${tag}" >> $GITHUB_OUTPUT
+          # Add matrix to .build/info to cache
+          make build/info/short > .build-info
+      - run: |
+          make images/${{ matrix.image }}
+      - run: |
+          make docker/save/${{ matrix.image }}
+      - name: Run container structure test
+        if: ${{ !contains(github.event.pull_request.labels.*.name, 'ci/skip-container-structure-test') && !contains(github.event.pull_request.labels.*.name, 'ci/skip-test') }}
+        run: |
+          make test/container-structure/${{ matrix.image }}
+      - uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
+        id: image-artifacts
+        with:
+          name: images_${{ matrix.image }}
+          path: |
+            ./build/docker/*.tar
+      # TODO Do we need to scan images for each arch?
+      - name: scan amd64 image
+        id: scan_image-amd64
+        uses: Kong/public-shared-actions/security-actions/scan-docker-image@b0ef627fa71528272d1daa9257b71dc90246cc46
+        with:
+          asset_prefix: image_${{ matrix.image }}-amd64
+          image: ./build/docker/${{ matrix.image }}-amd64.tar
+      - name: scan arm64 image
+        id: scan_image-arm64
+        if: ${{ fromJSON(inputs.FULL_MATRIX) }}
+        uses: Kong/public-shared-actions/security-actions/scan-docker-image@b0ef627fa71528272d1daa9257b71dc90246cc46
+        with:
+          asset_prefix: image_${{ matrix.image }}-arm64
+          image: ./build/docker/${{ matrix.image }}-arm64.tar
+        # TODO in the future we may want to have prerelease images and use `regctl image copy` to move them to their final location
+      - name: publish images
+        id: release_images
+        env:
+          DOCKER_API_KEY: ${{ secrets.DOCKER_API_KEY }}
+          DOCKER_USERNAME: ${{ vars.DOCKER_USERNAME }}
+        run: |-
+          make docker/login
+          # ensure we always logout
+          function on_exit() {
+            make docker/logout
+          }
+          trap on_exit EXIT
+          make docker/push/${{ matrix.image }}
+          make docker/manifest/${{ matrix.image }}
+      - name: Install regctl
+        uses: regclient/actions/regctl-installer@d8097ee5dd5cdf150516315919b58509fc7f4cfa
+      - name: image digest
+        id: image_digest
+        if: ${{ fromJSON(inputs.ALLOW_PUSH) }}
+        run: |
+          echo "Fetching image digest for ${{ matrix.image }}"
+          digest=$(regctl image digest ${{ steps.image_meta.outputs.image }})
+          echo "digest=${digest}" >> $GITHUB_OUTPUT
+      - name: sign image
+        # TODO At the it's asking for verifying a token in a browser... Something seems off
```

Comment:
@saisatishkarra I'm skipping signing because it was doing:


```
+ cosign sign --yes --recursive --tlog-upload=true -a repo=kumahq/kuma -a workflow=build-test-distribute -a sha=2287c4fbf6615b7de5f82a98c114c252ec0a4186 0.0.0-preview.v2287c4fbf@
Generating ephemeral keys...
Retrieving signed certificate...
Non-interactive mode detected, using device flow.
Enter the verification code WFGJ-LXVF in your browser at: https://oauth2.sigstore.dev/auth/device?user_code=WFGJ-LXVF
Code will be valid for 300 seconds
```

Is there something specific to do to not have to verify these? Or do we need to always have a human interaction to sign manifests?

Reply:
I think it might just be because of missing permissions:

https://github.com/Kong/public-shared-actions/tree/main/security-actions/sign-docker-image#workflow--job-permissions-for-keyless-oidc-signing

I'll keep the job as skipped for now and will look at fixing this in follow up PRs.